### DISCUSSION TITLE
set node's message on failure

### DISCUSSION THREAD

**Discussion by PuntoDiGoccia** [argoproj/argo-workflows#5708] [Q&A] [answered]
Doc states that `workflow.failure` has these available fields: `displayName`, `message`, `templateName`, `phase`, `podName`, and `finishedAt`
I'm confused: can `message` be set to a user-defined value on failure? how can it be done in a workflow?

**Comment by terrytangyuan** [comment, marked as answer]
Are you looking at this document? https://argoproj.github.io/argo-workflows/variables/#exit-handler

If so, these are read-only variables that can be referenced in the workflow. For instructions on how to use it, see https://argoproj.github.io/argo-workflows/variables/#how-to-use-variables

**Reply by PuntoDiGoccia** [reply]
> Are you looking at this document? https://argoproj.github.io/argo-workflows/variables/#exit-handler

yep

> If so, these are read-only variables that can be referenced in the workflow. For instructions on how to use it, see https://argoproj.github.io/argo-workflows/variables/#how-to-use-variables

ok, thanks!
I was looking for a way to exit a workflow at any step and print a customized error message explaining the reason of the failure in the exit handler; I thought that `message` could be set as per this CLI command https://argoproj.github.io/argo-workflows/cli/argo_node/

**Reply by terrytangyuan** [reply]
Yes you should be able to modify the message using the CLI but that specific document is for referencing the variables.

**Reply by PuntoDiGoccia** [reply]
good, I'll use `globalName` outputs then.
Thanks
